integration-webcert/src/test/java/se/inera/intyg/minaintyg/integration/webcert/converter/metadata/EventConverterTest.java

```diff
@@ -222,11 +222,46 @@ void shallReturnEmptyListIfReplacingCertificateIsRevoked() {
     assertEquals(Collections.emptyList(), actualEvents);
   }
 
+  @Test
+  void shallSortEventsOnCreationTimestamp() {
+    final var sent = LocalDateTime.now();
+    final var replaced = LocalDateTime.now().plusDays(1);
+
+    final var expectedEvents = List.of(
+        createEvent(REPLACES_DESCRIPTION, replaced),
+        CertificateEvent.builder()
+            .description("Skickat till recipientName")
+            .timestamp(sent)
+            .build()
+    );
+
+    final var metadataDTO = CertificateMetadataDTO.builder()
+        .recipient(CertificateRecipient.builder()
+            .name("recipientName")
+            .sent(sent)
+            .build())
+        .relations(
+            CertificateRelations.builder()
+                .parent(
+                    createRelation(expectedEvents.get(0), CertificateRelationType.COMPLEMENTED)
+                )
+                .build()
+        )
+        .build();
+
+    final var actualEvents = eventConverter.convert(metadataDTO);
+    assertEquals(expectedEvents, actualEvents);
+  }
+
   private static CertificateEvent createReplaceEvent(String description) {
+    return createEvent(description, TIMESTAMP);
+  }
+
+  private static CertificateEvent createEvent(String description, LocalDateTime timestamp) {
```

Comment:
If you overload the createEvent again you can add an option to pass a certificateId.